catalog/ui/search-ui/standard/src/main/webapp/js/view/Modal.js

```diff
@@ -27,13 +27,15 @@ define([
                 this.className = 'modal fade ' + this.className; // add on modal specific classes.
                 Marionette.LayoutView.prototype.constructor.apply(this, arguments);
             },
-            initialize: function () {
+            initialize: function (view, shouldDestroy) {
                 // destroy view after animation completes.
                 // extending views must call: Modal.prototype.initialize.apply(this, arguments);
-                var view = this;
                 this.$el.one('hidden.bs.modal', function () {
-                    view.destroy();
-                });
+                    if (view !== undefined && shouldDestroy ||
+                        shouldDestroy === undefined) {
```

Comment:
Not sure the conditional is quite right, you're never actually calling "shouldDestroy", should just be something like `if(shouldDestroy === undefined || shouldDestroy())` We shouldn't need to update the preferences modal to make this change.